### PR TITLE
 Add RTF_ADD_SUITE command

### PR DESCRIPTION
**Pull request by drdanz**
This command creates a suite xml file for each configuration and add a test that runs the created suite using testrunner.

```cmake
rtf_add_suite(NAME <name>
              [RTF_FIXTURE fixture1 [args1...]
              [RTF_FIXTURE fixture2 [args2...]] ...]
              [RTF_TEST test1 TYPE type [ARGS <args1>...
                                        [ARGS <args2>...] ...]
              [RTF_TEST test2 TYPE type [ARGS <args3>...] ...] ...]
              [CONFIGURATIONS <config>...]
              [WORKING_DIRECTORY <dir>]
              [WEB_REPORTER]
              [WEB_PORT <port>]
              [DETAIL]
              [VERBOSE]
              [OUTPUT <output>]
              [OUTPUT_TYPE <output_type>]
```

The suite contains one `fixture` line for each `FIXTURE` and one `test` for each `ARGS` in each `TEST`.
Both `FIXTURE` and `TEST` can be target names, and in that case they will be replaced with the full path to the module in each configuration.

`CONFIGURATIONS` and `WORKING_DIRECTORY` are forwarded to `add_test`

`WEB_REPORTER`, `WEB_PORT`, `DETAIL`, `VERBOSE`, `OUTPUT`, and `OUTPUT_TYPE` are passed to testrunner. See `testrunner --help` for details.

### CODE REVIEW

**Comment by drdanz** [comment]
Please note that it is untested on MSVC, please test it before merging

**Comment by aerydna** [comment]
the macro is under test on MSVC

**Comment by traversaro** [comment]
I had a problem with the TEST argument, fixed with https://github.com/drdanz/robot-testing/pull/1/files .

**Comment by traversaro** [comment]
The filename of the generated suite xml file contains the name of the test name, if the test name contains "::" (as in YARP) this is problematic in Windows.

**Comment by traversaro** [comment]
There are a few CMake warnings on CMake 3.9 : 
~~~
CMake Warning (dev) at /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:227 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted keywords like "TEST" will no longer be interpreted as keywords when
  the policy is set to NEW.  Since the policy is not set the OLD behavior
  will be used.
Call Stack (most recent call first):
  /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:322 (_rtf_parse_arguments_with_callback)
  tests/rtf/NetworkClock/CMakeLists.txt:8 (rtf_add_suite)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:227 (if):
  Policy CMP0064 is not set: Support new TEST if() operator.  Run "cmake
  --help-policy CMP0064" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  TEST will be interpreted as an operator when the policy is set to NEW.
  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:322 (_rtf_parse_arguments_with_callback)
  tests/rtf/NetworkClock/CMakeLists.txt:8 (rtf_add_suite)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:232 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted keywords like "TEST" will no longer be interpreted as keywords when
  the policy is set to NEW.  Since the policy is not set the OLD behavior
  will be used.
Call Stack (most recent call first):
  /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:322 (_rtf_parse_arguments_with_callback)
  tests/rtf/NetworkClock/CMakeLists.txt:8 (rtf_add_suite)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:232 (if):
  Policy CMP0064 is not set: Support new TEST if() operator.  Run "cmake
  --help-policy CMP0064" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  TEST will be interpreted as an operator when the policy is set to NEW.
  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:322 (_rtf_parse_arguments_with_callback)
  tests/rtf/NetworkClock/CMakeLists.txt:8 (rtf_add_suite)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:236 (elseif):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted keywords like "TEST" will no longer be interpreted as keywords when
  the policy is set to NEW.  Since the policy is not set the OLD behavior
  will be used.
Call Stack (most recent call first):
  /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:322 (_rtf_parse_arguments_with_callback)
  tests/rtf/NetworkClock/CMakeLists.txt:8 (rtf_add_suite)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:322 (_rtf_parse_arguments_with_callback):
  Policy CMP0064 is not set: Support new TEST if() operator.  Run "cmake
  --help-policy CMP0064" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  TEST will be interpreted as an operator when the policy is set to NEW.
  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  tests/rtf/NetworkClock/CMakeLists.txt:8 (rtf_add_suite)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:239 (elseif):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted keywords like "TEST" will no longer be interpreted as keywords when
  the policy is set to NEW.  Since the policy is not set the OLD behavior
  will be used.
Call Stack (most recent call first):
  /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:322 (_rtf_parse_arguments_with_callback)
  tests/rtf/NetworkClock/CMakeLists.txt:8 (rtf_add_suite)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:322 (_rtf_parse_arguments_with_callback):
  Policy CMP0064 is not set: Support new TEST if() operator.  Run "cmake
  --help-policy CMP0064" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  TEST will be interpreted as an operator when the policy is set to NEW.
  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  tests/rtf/NetworkClock/CMakeLists.txt:8 (rtf_add_suite)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:242 (elseif):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted keywords like "TEST" will no longer be interpreted as keywords when
  the policy is set to NEW.  Since the policy is not set the OLD behavior
  will be used.
Call Stack (most recent call first):
  /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:322 (_rtf_parse_arguments_with_callback)
  tests/rtf/NetworkClock/CMakeLists.txt:8 (rtf_add_suite)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:322 (_rtf_parse_arguments_with_callback):
  Policy CMP0064 is not set: Support new TEST if() operator.  Run "cmake
  --help-policy CMP0064" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  TEST will be interpreted as an operator when the policy is set to NEW.
  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  tests/rtf/NetworkClock/CMakeLists.txt:8 (rtf_add_suite)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:187 (elseif):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted keywords like "TEST" will no longer be interpreted as keywords when
  the policy is set to NEW.  Since the policy is not set the OLD behavior
  will be used.
Call Stack (most recent call first):
  /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:228 (_rtf_add_suite_cb)
  /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:322 (_rtf_parse_arguments_with_callback)
  tests/rtf/NetworkClock/CMakeLists.txt:8 (rtf_add_suite)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:228 (_rtf_add_suite_cb):
  Policy CMP0064 is not set: Support new TEST if() operator.  Run "cmake
  --help-policy CMP0064" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  TEST will be interpreted as an operator when the policy is set to NEW.
  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  /Users/traversaro/src/isaac-superbuild/build/install/lib/cmake/RTF/RTFConfig.cmake:322 (_rtf_parse_arguments_with_callback)
  tests/rtf/NetworkClock/CMakeLists.txt:8 (rtf_add_suite)
This warning is for project developers.  Use -Wno-dev to suppress it.
~~~

File: .appveyor.yml [hunk 1]
```diff
@@ -14,11 +14,16 @@ configuration:
...
   matrix:
-    - CTEST_OUTPUT_ON_FAILURE: 1
-      BUILD_SHARED_LIBS: ON
+    - BUILD_SHARED_LIBS: ON
+    - BUILD_SHARED_LIBS: OFF
 
-    - CTEST_OUTPUT_ON_FAILURE: 1
+matrix:
+  exclude:
+    - image: Visual Studio 2013
```
Comment (traversaro) [inline .appveyor.yml]: <3

**Comment by Nicogene** [comment]
Merged thanks!
cc @aerydna

**Pull request merged**